### DISCUSSION TITLE
Does Turborepos dependency graph work with webpack’s Module Federation?

### DISCUSSION THREAD

**Discussion by florianmatz** [vercel/turborepo#589] [Help] [answered]
Hi guys :)

Let‘s assume that we wanted to build a AppShell that serves different Microfrontends.

The AppShell itself is also a MF, all of them live in the apps folder.

The AppShell aquires all of the Microfrontends via webpack’s Module Federation plugin.

If you now make a change to one of the Microfrontends would Turborepo consider this and run i.e. tests for the affected parts in the shell or vice versa if the AppShell would expose a component via ModuleFederation and that component changed?

**Comment by anthonyshew** [comment, marked as answer]
You would need to make the build scripts aware of the federated modules through the dependency graph or `inputs`.
